pkg/lifecycle/once.go

```diff
@@ -230,11 +222,11 @@ func (l *lifecycleOnce) loadError() error {
 // start to full stop.
 // The function only guarantees that the lifecycle has at least passed through
 // the returned state and may have progressed further in the intervening time.
-func (l *lifecycleOnce) LifecycleState() LifecycleState {
-	return LifecycleState(l.state.Load())
+func (o *Once) LifecycleState() State {
```

Comment:
Actually, will you never embed this type if you're implementing Lifecycle because the `Start` and `Stop` functions will collide with `Start` and `Stop` for the `Lifecycle`?